app/repositories/SessionRepository.scala

```diff
@@ -67,22 +70,28 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
     )
   }
 
-  def get(key: String): Future[Option[Process]] = {
+  def get(key: String): Future[RequestOutcome[Process]] = {
     val updateModifier = Json.obj("$set" -> Json.obj("lastAccessed" -> Json.obj("$date" -> DateTime.now(DateTimeZone.UTC).getMillis)))
     findAndUpdate(Json.obj("_id" -> key), updateModifier, fetchNewObject = false)
-      .map(_.result[DefaultSessionRepository.SessionProcess].map(_.process))
+      .map(wr => wr.result[SessionProcess].fold(Left(DatabaseError): RequestOutcome[Process])(r => Right(r.process)))
       .recover {
         case lastError =>
           logger.error(s"Unable to retrieve process associated with key=$key")
-          None
+          Left(DatabaseError)
       }
   }
 
-  def set(key: String, process: Process): Future[Option[Unit]] =
-    insert(DefaultSessionRepository.SessionProcess(key, process.meta.id, process, DateTime.now(DateTimeZone.UTC))) map (_ => Some(())) recover {
-      case lastError =>
-        logger.error(s"Unable to persist process=${process.meta.id} against id=$key")
-        None
-    }
+  def set(key: String, process: Process): Future[RequestOutcome[Unit]] = {
+    val sessionDocument = Json.toJson(DefaultSessionRepository.SessionProcess(key, process.meta.id, process, DateTime.now(DateTimeZone.UTC)))
```

Comment:
Can you use the type SessionProcess here?

Reply:
yes?

Reply:
Ok, I would say if you've gone to the trouble of creating a special type you should use it wherever necessary.

Reply:
Type declaration removed.